src/lib/Scenes/Map/Bucket.ts

```diff
@@ -1,22 +1,36 @@
 import { GlobalMap_viewer } from "__generated__/GlobalMap_viewer.graphql"
 import { filter } from "lodash"
+import moment from "moment"
 import { Tab } from "./types"
 
-export type BucketKey = "saved" | "fairs" | "galleries" | "museums"
+export type BucketKey = "saved" | "fairs" | "galleries" | "museums" | "closing" | "opening"
 export type BucketResults = { [key in BucketKey]: any[] }
 
 export const bucketCityResults = (viewer: GlobalMap_viewer): BucketResults => {
-  // TODO: Saved isn't supported yet, so we'll return nothing.
-  const saved = []
-
-  const fairs = (viewer.city.fairs as unknown) as Tab[]
+  const saved = filter(viewer.city.shows.edges, e => e.node.is_followed === true)
+  const oneWeekFromNow = moment(new Date()).add(1, "week")
```

Comment:
#minor i wonder if we should make these constants somewhere, to define the thresholds for upcoming and closing.